src/hypertable.c

```diff
@@ -954,7 +955,7 @@ hypertable_create(PG_FUNCTION_ARGS)
 				 errmsg("table \"%s\" is not empty", get_rel_name(table_relid)),
 				 errhint("You can migrate data by specifying 'migrate_data => true' when calling this function.")));
 
-	if (find_inheritance_children(table_relid, AccessShareLock) != NIL)
+	if (find_inheritance_children(table_relid, AccessShareLock) != NIL || inheritance_parent_relid(table_relid) != InvalidOid)
```

Comment:
Wrap this new function in a macro or other function: `is_inheritance_child()` or `has_inheritance_parent()` for clarity.

Then maybe the complete thing in `table_uses_inheritance()` or similar?